include/curl/curl.h

```diff
@@ -35,7 +35,7 @@
 
 /* Compile-time deprecation macros. */
 #if defined(__GNUC__) && defined(__GNUC_MINOR__) && \
-    ((__GNUC__ > 5) || (__GNUC__ == 5 && __GNUC_MINOR__ >= 3)) && \
+    ((__GNUC__ > 6) || (__GNUC__ == 6 && __GNUC_MINOR__ >= 1)) && \
```

Comment:
I think gcc 6.1 was the first version of the gcc 6.x releases (there was no gcc 6.0)

Maybe check the version like this:
```
#if defined(__GNUC__) && (__GNUC__ >= 6) && \
    !defined(__INTEL_COMPILER) && \
    !defined(CURL_DISABLE_DEPRECATION) && !defined(BUILDING_LIBCURL)
```